mediorum/server/repair.go

```diff
@@ -94,23 +94,37 @@ func (ss *MediorumServer) repairUnderReplicatedBlobs() error {
 
 	for _, problem := range problems {
 
+		logger := logger.With("cid", problem.Key)
+
 		// if I have the file...
 		// call replicate
 		iHave := strings.Contains(problem.Hosts, ss.Config.Self.Host)
 		if iHave {
 			logger.Info("repairing", "cid", problem.Key)
-			blob, err := ss.bucket.NewReader(ctx, problem.Key, nil)
+			r, err := ss.bucket.NewReader(ctx, problem.Key, nil)
+			if err != nil {
+				logger.Error("failed to read key", err)
+				continue
+			}
+			defer r.Close()
+
+			// validate CID matches
+			err = validateCID(problem.Key, r)
 			if err != nil {
-				logger.Warn("failed to read key", "key", problem.Key, "err", err)
+				logger.Error("CID validation failed", err)
```

Comment:
nit: would it be good to keep the cid field here for the searchability you mentioned?

Reply:
It gets added to the logger instance a few lines up:
`logger := logger.With("cid", problem.Key)`

Reply:
ah right, very nice